pandas/io/parsers.py

```diff
@@ -757,6 +757,8 @@ def read_table(
     memory_map=False,
     float_precision=None,
 ):
+    if delim_whitespace and sep == "\t":
+        sep = ","
```

Comment:
because it is not immediately obvious why this is needed, fixing the signature to use lib.no_default in this PR may be best or add a comment here